`Development`: Add documentation for lectures (#5373)
ls1intum/Artemis: docs/user/lectures.rst

.. _lectures:

Lectures
========

.. contents:: Content of this document
    :local:
    :depth: 2

Overview
--------
Apart from :ref:`Exercises` and :ref:`Exams`, Artemis can also host the **Lectures** of a course.

Students
^^^^^^^^

> Students can see all published lectures of a course, grouped by date, by clicking on the tab labeled *Lectures*.

Reviewer: "by" twice
```suggestion
Students can see all published lectures of a course, grouped by date, when clicking on the tab labeled *Lectures*.
```

|lectures-overview|

> Clicking on a lecture leads to following page, where students can see the title, description, lecture units, and attachments.

Reviewer 2: ```suggestion
Clicking on a lecture leads to the following page, where students can see the title, description, lecture units, and attachments.
```

|lecture-details|

Instructors
^^^^^^^^^^^

> Instructors can upload lecture slides, divide lectures into units, integrate video streams, lecture recordings, and exercises into lectures, and define learning goals.

Reviewer 3: ```suggestion
Instructors can upload lecture slides, divide lectures into units, integrate video streams, lecture recordings, and exercises into lectures and define learning goals.
```

Reviewer 4: ```suggestion
Instructors can upload lecture slides, integrate video streams, and lecture recordings.
This material can be grouped into logical lecture units.
To directly link the the necessary knowledge to its application, regular course exercises can be made part of lectures, too.
Instructors can also define learning goals so that students can easier keep track of the knowledge they should have after working with those lecture materials.
```

I think it would be worth elaborating more on that single sentence anyway as it combines a lot of different elements of lectures into one.
The suggestion is just a quick write-up of the four main elements that I suppose the sentence tries to convey. I’m sure they can be extended or reworded in a better way.

On the course management site, clicking on |lectures-management-btn| opens the following page for managing lectures.

|lectures-management|

> Lectures

Reviewer: Maybe "Creating Lectures" or "Managing Lectures"? This section is only about how instructors can work with them.

--------

Instructors have two options for creating a lecture.

    1. Create a new lecture from scratch by clicking on |create-lecture-btn|.
    Lectures consist of a title, a description, and optionally a start and end date.

        |create-lecture|

    2. Alternatively, instructors can also import a lecture from a any other course where they at least have *editor* access.
    Clicking on |import-lecture-btn| opens the import modal, where instructors search for an exisiting lecture and import it.

        |import-lecture|

Once a lecture is created, instructors can add attachments to it.
An attachment is a file (e.g., document, image) with a title and an optional release date.

|create-attachment|

Lecture Units
-------------

Lectures can be divided into lecture units, which can be of the following types:

    * **Text unit:** A text with markup.
    * **Exercise unit:** An exercise from the same course.

>     * **Video unit:** An embedded video stream or video from an external source (e.g., RGB live steam, YouTube video).

Reviewer: Should we even mention the RBG here? I think it would be better to just mention live streams in general.
```suggestion
    * **Video unit:** An embedded video stream or video from an external source (e.g., RBG live steam, YouTube video).
```

Online Unit
^^^^^^^^^^^
An online unit consists of a link to an external website, a name, and optionally a description and release date.
Artemis automatically pre-fills the title and description from the website's metadata once the URL is set.
Students complete this unit once they navigate to the external website.

|create-online-unit|

Text Unit
^^^^^^^^^
An text unit consists of a name, some rich-text content and optionally a release date.
Students complete this unit as soon as they open it.

|create-text-unit|

Video Unit
^^^^^^^^^^^
An video unit consists of a name, an embeddable video link, and optionally a description and release date.
Artemis can convert the website link from common video sources (RGB, YouTube, Vimeo) to an embeddable URL using the arrow button.
Students complete this unit when they watch the video for at least five minutes.

|create-video-unit|

Learning Goals
--------------

Instructors can define learning goals to group course content and associate units from different lectures with the same learning goal.

|learning-goals-management|

To create a learning goal, as an instructor click on the plus button and enter a name and description.

> Then, select a lecture from the dropdown and click on its units to associate them with this learning goal.

Reviewer: I think this comma is wrong, but I'm not sure.
```suggestion
Then select a lecture from the dropdown and click on its units to associate them with this learning goal.
```
